orthoinference/src/main/java/org/reactome/orthoinference/ReactionInferrer.java

```diff
@@ -52,87 +51,72 @@ public static void inferReaction(GKInstance reactionInst) throws Exception
-				String eligibleEventName = reactionInst.getAttributeValue(DB_ID).toString() + "\t" + reactionInst.getDisplayName() + "\n";	
-				// Having passed all tests/filters until now, the reaction is recorded in the 'eligible reactions' file, meaning inference is continued.
-				eligibleCount++;
-				Files.write(Paths.get(eligibleFilehandle), eligibleEventName.getBytes(), StandardOpenOption.APPEND);
-				// Attempt to infer all PhysicalEntities associated with this reaction's Input, Output, CatalystActivity and RegulatedBy attributes.
-				// Failure to successfully infer any of these attributes will end inference for this reaction.
-				logger.info("Inferring inputs...");
-				if (inferReactionInputsOrOutputs(reactionInst, infReactionInst, input))
+				logger.info("Inferring outputs...");
+				if (inferReactionInputsOrOutputs(reactionInst, infReactionInst, output))
 				{
-					logger.info("Inferring outputs...");
-					if (inferReactionInputsOrOutputs(reactionInst, infReactionInst, output))
+					logger.info("Inferring catalysts...");
+					if (inferReactionCatalysts(reactionInst, infReactionInst))
 					{
-						logger.info("Inferring catalysts...");
-						if (inferReactionCatalysts(reactionInst, infReactionInst))
+						// Many reactions are not regulated at all, meaning inference is attempted but will not end the process if there is nothing to infer.
+						// The inference process will end though if inferRegulations returns an invalid value.
+						logger.info("Inferring regulations...");
+						List<GKInstance> inferredRegulations = inferReactionRegulations(reactionInst);
+						if (inferredRegulations.size() == 1 && inferredRegulations.get(0) == null)
 						{
-							// Many reactions are not regulated at all, meaning inference is attempted but will not end the process if there is nothing to infer. 
-							// The inference process will end though if inferRegulations returns an invalid value.
-							logger.info("Inferring regulations...");
-							List<GKInstance> inferredRegulations = inferReactionRegulations(reactionInst);
-							if (inferredRegulations.size() == 1 && inferredRegulations.get(0) == null)
-							{
-								return;
-							}
-							if (infReactionInst.getSchemClass().isValidAttribute(releaseDate)) 
-							{
-								infReactionInst.addAttributeValue(releaseDate, dateOfRelease);
-							}
-							// FetchIdenticalInstances would just return the instance being inferred. Since this step is meant to always
-							// add a new inferred instance, the storeInstance method is just called here.
-							GKInstance orthoStableIdentifierInst = EventsInferrer.getStableIdentifierGenerator().generateOrthologousStableId(infReactionInst, reactionInst);
-							infReactionInst.addAttributeValue(stableIdentifier, orthoStableIdentifierInst);
-							dba.storeInstance(infReactionInst);
-							logger.info("Inferred RlE instance: " + infReactionInst);
+							return;
```

Comment:
Is this an if-statement that only contains a `return`? Could this be modified to be the _negation_ of the same condition which _allows_ the rest of the code to flow? This is a void function, so it's not even expected to return anything. I find that using `return` as a form of flow-control to be confusing.